src/backend/migrate-ipfs-repo-if-necessary.js

```diff
@@ -0,0 +1,37 @@
+'use strict'
+
+module.exports = () => {
+  return new Promise((resolve, reject) => {
+    const indexedDB = window.indexedDB || window.mozIndexedDB || window.webkitIndexedDB || window.msIndexedDB || window.shimIndexedDB;
+    var open = indexedDB.open('ipfs');
+    open.onerror = reject
+    open.onsuccess = () => {
+      const db = open.result
+      const stores = db.objectStoreNames
+      if (!stores.contains('ipfs')) {
+        // no IPFS db, no need to upgrade
+        return resolve()
+      }
+      const tx = db.transaction('ipfs', 'readwrite');
+      const store = tx.objectStore('ipfs');
+      const get = store.get('/version')
+      get.onerror = reject
+      get.onsuccess = () => {
+        if (!get.result) {
+          return resolve()
+        }
+        const version = Number(Buffer.from(get.result).toString())
+        if (version === 5) {
+          console.log('saving new version')
+          const put = store.put(new Buffer('6'), '/version')
```

Comment:
I'm missing something. How can this work if there was a new version? Has to be some backwards-incompatible changes in that case that the migration should handle, or there shouldn't be a version change

Reply:
@VictorBjelkholm the IPFS Repo story is a complicated one. tl;dr; There is no spec, we worked hard to make our implementation interoperable with go-ipfs, go-ipfs changed its repo (more than one time), things kept breaking and js-ipfs tried to catch up to the point that it is just impossible to follow something without notice. Today we are still interoperable go-ipfs -> js-ipfs https://github.com/ipfs/js-ipfs/blob/master/test/interop/repo.js, but the other way it doesn't work because go-ipfs expects some new values for Datastore that are not speced out. 

Although the repo has changed for go-ipfs, for js-ipfs is the same because we don't have things like badger.